### ISSUE TITLE
Jackson deserialization crash in PublicKeyCredential.parseRegistrationResponseJson() with Chrome 99.0.4844.82 on Ubuntu 20.04.4 LTS

### ISSUE THREAD

**Issue by toby1984** [Yubico/java-webauthn-server#175]
Hi,

For me, Jackson deserialization is crashing because the received JSON contains an unrecognized "authenticatorAttachment": "cross-platform" property.

This used to work so I assume some recent update (either a Chrome or OS update) changed the behaviour.

The crash is happening with 

 - JDK17 
 - Chrome 99.0.4844.82
 - Ubuntu 20.04.04 LTS
 - U2F Yubikey (oldish Yubikey key, the blue USB-A one without NFC etc)
 - java-webauthn-server 1.12.2 
 
Note: I removed the Bas64-encoded payload from the stack trace as I'm not sure what is ok to share publicly.

 ```
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "authenticatorAttachment" (class com.yubico.webauthn.data.PublicKeyCredential), not marked as ignorable (5 known properties: "response", "clientExtensionResults", "id", "rawId", "type"])
 at [Source: (String)"{"response":{"attestationObject":"..........","clientDataJSON":"........"[truncated 244 chars]; line: 1, column: 745] (through reference chain: com.yubico.webauthn.data.PublicKeyCredential["authenticatorAttachment"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1127) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1989) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1700) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperties(BeanDeserializerBase.java:1650) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:539) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3612) ~[jackson-databind-2.13.1.jar:2.13.1]
	at com.yubico.webauthn.data.PublicKeyCredential.parseRegistrationResponseJson(PublicKeyCredential.java:202) ~[webauthn-server-core-minimal-1.12.2.jar:1.12.2]
```
![json](https://user-images.githubusercontent.com/271364/160113721-9ec031d6-23cc-4ab4-97eb-6da8e3d70a96.png)

**Comment by toby1984** [comment]
For the time being I work around this problem by just doing

            if ( json.contains("\"authenticatorAttachment\"" ) ) {
                json = json.replaceAll( "\"authenticatorAttachment\"\s*:\s*\".*?\"\s*,?", "" );
            }

but this is obviously not ideal.

**Comment by emlun** [comment]
Fixed in version `1.12.3-RC1`, available now on Maven Central (though it hasn't shown up on the indexed website yet). We'll promote this to a proper release in a couple of weeks, probably along with a few other upcoming things.

Thanks again!
